modules/xmpp/xmpp.js

```diff
@@ -233,13 +224,13 @@ export default class XMPP extends Listenable {
             // FIXME no need to do it again on stream resume
             this.caps.getFeaturesAndIdentities(pingJid)
                 .then(({ features, identities }) => {
-                    if (features.has(Strophe.NS.PING)) {
-                        this._pingSupported = true;
-                        this.connection.ping.startInterval(pingJid);
-                    } else {
-                        logger.warn(`Ping NOT supported by ${pingJid}`);
+                    if (!features.has(Strophe.NS.PING)) {
+                        logger.error(
+                            `Ping NOT supported by ${pingJid} - please enable ping in your XMPP server config`);
                     }
 
+                    this.connection.ping.startInterval(pingJid);
+
```

Comment:
So we start the ping even if it is not supported?

Reply:
Yes, because it's going to be mandatory so it can be used to kill a zombie WebSocket early. If it starts that will keep on brining error in the console and make it harder to be missed.

Reply:
it's also used to simplify the flow around ice failed handling

Reply:
I've added extra comment